cmd/podman/patch.go

```diff
@@ -0,0 +1,203 @@
+
+        rootCmd.AddCommand(patchCommand.Command)
+}
+
+// Define the entrypoint command
+func patchCmd(c *cliconfig.PatchValues) error {
+//func patchCmd(c *cli.Context) error {
+    runtime, err := libpodruntime.GetRuntime(&c.PodmanCommand)
+    //runtime, err := libpodruntime.GetRuntime(c)
+    if os.Geteuid() != 0 {
+        rootless.SetSkipStorageSetup(true)
+    }
+
+    if os.Geteuid() != 0 {
+        if driver := runtime.GetConfig().StorageConfig.GraphDriverName; driver != "vfs" {
+            // Do not allow to mount a graphdriver that is not vfs if we are
+            // creating the userns as part
+            // of the mount command.
+            fmt.Errorf("cannot mount using driver %s in rootless mode", driver)
+        }
+
+        became, ret, err := rootless.BecomeRootInUserNS()
+        if err != nil {
+            return err
+        }
+        if became {
+            os.Exit(ret)
+        }
+    }
+    args := c.InputArgs
+    if len(args) < 3 {
+        if c.All && len(args) < 2 {
+            return errors.Errorf("too few given arguments")
+        }
+    }
+
+    if err != nil {
+        return errors.Wrapf(err, "could not get runtime")
+    }
+    defer runtime.Shutdown(false)
+
+    dst := args[0]
+    patch := args[1]
+
+    var containers []string
+    if ! c.All {
+        containers = args[2:]
+    } else {
+        containers = retrieveExisting(runtime)
+    }
+
+    if ! canPatch() {
+        fmt.Println("Patch command not on your system")
+        os.Exit(1)
+    }
+
+    if ! exists(patch) {
+        fmt.Println("Patch file not found:", patch)
+        os.Exit(1)
+    }
+
+    return patchContainers(runtime, patch, dst, containers, c.Ignore)
+}
+
+// Retrieve all running containers and return array of name
+func retrieveExisting(runtime *libpod.Runtime) ([]string) {
+    runningPods, err := runtime.GetRunningContainers()
+    if err != nil {
+        fmt.Println(err)
+        os.Exit(1)
+    }
+    var containers []string
+
+    for _, container := range runningPods {
+        containers = append(containers, container.Name())
+    }
+    return containers
+}
+
+// Check if the patch command is available on the current system (host)
+func canPatch() (bool) {
+    _, err := exec.Command("which", "patch").Output()
+    if err != nil {
+        return false
+    }
+    return true
+}
+
+// Check if a file exist
+func exists(path string) (bool) {
+    _, err := os.Stat(path)
+    if os.IsNotExist(err) {
+        return false
+    }
+    if err == nil { return true }
+    return true
+}
+
+// Retrieve all the containers by name and
+// loop over to apply the given patch
+func patchContainers(runtime *libpod.Runtime,
+                     patch string, dst string, containers []string, ignore bool) error {
+    var ctrs []*libpod.Container
+    for _, container := range containers {
+        ctr, err := runtime.LookupContainer(container)
+        if err != nil {
+            fmt.Println(err)
+            os.Exit(1)
```

Comment:
im thinking you might need to break this func up so you can return your way out of it